src/float/cmp.rs

```diff
@@ -3,6 +3,12 @@
 use crate::float::Float;
 use crate::int::MinInt;
 
+#[cfg(target_arch = "avr")]
+type CmpResult = i8;
+
+#[cfg(not(target_arch = "avr"))]
+type CmpResult = i32;
```

Comment:
In https://github.com/Rahix/avr-hal/discussions/641#discussioncomment-12508954 you mentioned that this should be i16, but here it is i8. If `i16` is correct, could these just use `core::ffi::c_int`?

Reply:
Unfortunately it's not `c_int` - that's how LLVM defines it:

https://github.com/llvm/llvm-project/blob/1e6ba3cd2fe96be00b6ed6ba28b3d9f9271d784d/compiler-rt/lib/builtins/fp_compare_impl.inc#L22

Reply:
Thanks, looks like GCC has the same https://github.com/gcc-mirror/gcc/blob/16065b6239aab2eecfd7a50f58d38324ee6478ba/libgcc/config/avr/avr-lib.h#L25C17-L25C23.